### ISSUE TITLE
TextInputLayout

### ISSUE THREAD

**Issue by rooparsh** [material-components/material-components-android#1608]
**Description:** The suffix text has gravity top and there's no way we can change the gravity. It behaves like superscript text.

**Expected behavior:** It should behave liek what's mentioned in https://material.io/components/text-fields#filled-text-field.

**Source code:** .
``` xml
 <com.google.android.material.textfield.TextInputLayout
        android:id="@+id/til_weight"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginStart="@dimen/dimen_24dp"
        android:layout_marginTop="@dimen/dimen_24dp"
        app:boxBackgroundColor="@android:color/transparent"
        app:hintEnabled="false"
        app:layout_constraintStart_toEndOf="@id/iv_decrement"
        app:layout_constraintTop_toBottomOf="@id/tv_label"
        app:suffixText="kg"
        app:suffixTextAppearance="?attr/textAppearanceSubtitle1"
        >

        <com.google.android.material.textfield.TextInputEditText
          android:id="@+id/et_weight"
          android:layout_width="wrap_content"
          android:layout_height="wrap_content"
          android:fontFamily="@font/montserrat_medium"
          android:hint="000.0"
          android:inputType="numberDecimal"
          android:maxLength="8"
          android:maxLines="1"
          android:padding="@dimen/dimen_4dp"
          android:text="000.0"
          android:textSize="@dimen/text_24sp"
          tools:text="000.0"
          />
```

<img width="248" alt="Screenshot 2020-08-12 at 7 32 52 AM" src="https://user-images.githubusercontent.com/7676782/89967017-09f84400-dc6e-11ea-82ae-830cf1cf0a14.png">

**Android API version:** 30

**Material Library version:** 1.2.0

**Comment by rooparsh** [comment]
It seems that the gravity of suffixTextView is not readjusted after textSize changes

**Comment by gabrielemariotti** [comment]
The problem in your case is that `suffixTextView` and the `TextInputEditText` use different textsize.

<img width="381" alt="Schermata 2020-08-12 alle 09 32 32" src="https://user-images.githubusercontent.com/2583078/89988427-d4f4ed00-dc7f-11ea-89d3-cac47e6dd2a8.png">

I am not sure that it is an unexpected result.

**Comment by rooparsh** [comment]
@gabrielemariotti we are trying to achieve this kind of UI. It works with same textSize, but we are looking towards different text sizes. 

As a workaround, I tried on the fork from repo and it worked, but not sure if its general use case or our use case only. So haven't raised any PR for that. 

<img width="311"  alt="Screenshot 2020-08-12 at 1 44 06 PM" src="https://user-images.githubusercontent.com/7676782/89991786-f6ff6700-dca1-11ea-8f15-55bbb24b3bbc.png">

**Comment by matpag** [comment]
If you test on `1.3.0-alpha02` is it working?

**Comment by rooparsh** [comment]
Yes I tried 1.3.0 alpha 2 also. It didn't work. I added some methods in my fork to make it work

**Comment by gabrielemariotti** [comment]
@rooparsh It could be interesting if you post your code. It could be adapt for a general use case.

**Comment by rooparsh** [comment]
@gabrielemariotti upon inspection I found that when we set edit text and change the text size, the height of the edit text gets adjusted accordingly but the baseline of suffixTextView remains there itself, what was calculated in the initial

**Comment by em-soft** [comment]
still no fix after all this time?

**Comment by guanshuwen** [comment]
2022.09.29.  Any fix now? Set an extra TextView and align it to the EditText sounds awful.

**Comment by guanshuwen** [comment]
2022.09.29.  Any fix now? Set an extra TextView and align it to the EditText sounds awful.
